modules/nlu/src/views/full/entities/EntityNameModal.tsx

```diff
@@ -43,33 +43,31 @@ export const EntityNameModal: FC<Props> = props => {
     e.preventDefault()
 
     if (props.action === 'create') {
-      onCreateEntity()
+      await onCreateEntity()
     } else if (props.action === 'duplicate') {
-      onDuplicateEntity()
+      await onDuplicateEntity()
     } else if (props.action === 'rename') {
-      onRenameEntity()
+      await onRenameEntity()
     }
 
     props.closeModal()
   }
 
-  const onCreateEntity = () => {
+  const onCreateEntity = async () => {
     const entity = {
       id: getEntityId(name),
       name: name.trim(),
       type: type as NLU.EntityType,
       occurrences: []
     }
-    props.api.createEntity(entity).then(() => {
-      props.onEntityModified(entity)
-    })
+    props.onEntityModified(await props.api.createEntity(entity))
```

Comment:
Does props.api.createEntity returns the created entity ? Because previously it passed the same entity object to both methods

Reply:
Good catch, you are absolutely right!